File: ass5_go/converter/converter.go
package converter

import (
	// "fmt"
    "ass5_go/stack"
    "ass5_go/queue"
    "ass5_go/valid"
)


func InfixToPostfix(query string) *queue.QueueUsingSlices {

    st := &stack.StackUsingSlices{}
	result := &queue.QueueUsingSlices{}

    for i := 0; i < len(query); i++ {
        var ch rune = rune(query[i])
        
        if ch == '(' {
            st.Push("(")
        } else if ch == ')' {
            for st.Peek().(string) != "(" {
                result.Push(st.Pop().(string))
            }
            st.Pop()
        } else if ch == '+' || ch == '-' || ch == '*' || ch == '/' {
            for st.Len() != 0 && valid.Precedence(ch) <= valid.Precedence(rune(st.Peek().(string)[0])) {
                result.Push(st.Pop().(string))
            }
            var ss string
            ss += string(ch)
            st.Push(ss)
        } else {
            var temp string
            for i < len(query) && valid.Isnum(rune(query[i])) {
                temp += string(query[i])
                i++
            }
            i--
            result.Push(temp)
        }
    }
    for st.Len() !=  0 {
        result.Push(st.Pop().(string))
    }
    return result
}

File: Ass5_python/python/include/converter.py
import validations as vd
import queue

class Converter(vd.Validations):
    def __init__(self, str):
        super().__init__()
        self.result = queue.Queue()
        self.ec = vd.Validations()
        if not self.ec.isValid(str):
            print("invalid")

    def infixToPostfix(self, query):
        st = queue.LifoQueue()
        n = len(query)
        for i in range(n):
            ch = query[i]
            if ch == '(':
                st.put("(")
            elif ch == ')':
                val = st.get()
                while val != "(":
                    self.result.put(val)
                    st.get()
            elif ch == '+' or ch == '-' or ch == '*' or ch == '/':
                if not st.empty():
                    val = st.get()
                    st.put(val)
                while not st.empty() and (self.ec.precedence(ch) <= self.ec.precedence(val[0])): 
                    self.result.put(st.get())
                ss = ""
                ss += ch
                st.put(ss)
            else:
                temp = ""
                while i < n and (query[i] >= '0' and query[i] <= '9' or query[i] == '.'):
                    temp += query[i]
                    i+=1
                i-=1
                self.result.put(temp)
            # print("2")
        while not st.empty():
            self.result.put(st.get())

        return self.result